classes/plugins/load.py

```diff
@@ -0,0 +1,75 @@
+from multiprocessing import Array
+from . import api
+from . import event
+
+import os
+import logging
+import inspect
+
+def _getPlugins_() -> Array:
+    import pkgutil
+    import importlib
+    if not (os.path.exists('plugins/')):
+        os.mkdir('plugins/')
+    import plugins as exts
+
+    def unqualify(name: str) -> str:
+        """Return an unqualified name given a qualified module/package `name`."""
+        return name.rsplit(".", maxsplit=1)[-1]
+    plugins = {}
+    for module in pkgutil.walk_packages(exts.__path__, f"{exts.__name__}."):
+        plugin_name = unqualify(module.name)
+        if plugin_name.startswith("_"):
+            next
+        plugins[plugin_name] = importlib.import_module(module.name)
+    order = {}
+    for plugin_name in plugins:
+        plugin = plugins[plugin_name]
+        try:
+            tmp = order[int(plugin.PRIORITY)]
+        except KeyError:
+            order[int(plugin.PRIORITY)] = []
+        except AttributeError:
+            plugin.PRIORITY = 999
+        order[int(plugin.PRIORITY)].append(plugin)
+    max = 0
+    for p in order:
+        plug = order[p]
+        for i in plug:
+            if int(i.PRIORITY) > max:
+                max = int(i.PRIORITY)
+    ret = []
+    for i in range(max+1):
+        ret.append(None)
+    for p in order:
+        ret[p] = order[p]
+    return ret
+
+def load_plugins():
+    event._reset_()
+    plugins = _getPlugins_()
+    logging.debug(plugins)
+    for index in plugins:
+        if index != None:
+            for imported in index:
+                if inspect.isfunction(getattr(imported, "load", None)):
+                    imported.load()
```

Comment:
Don't check if `load` is callable, just call it, and if you get an error then you should handle appropriately (i.e., log and move on).

Reply:
yeah but i was told no not rely on errors because it could be called by something else which would mean it would log the wrong message

Reply:
sorry could* not would